common/tokensParser.go

```diff
@@ -29,23 +34,20 @@ func ExtractTokenIDAndNonceFromTokenStorageKey(tokenKey []byte) ([]byte, uint64)
 		return tokenKey, 0
 	}
 
-	indexOfTokenHyphen := getIndexOfTokenHyphen(token, indexOfFirstHyphen)
-	tokenTicker := token[:indexOfTokenHyphen]
-	randomSequencePlusNonce := token[indexOfTokenHyphen+1:]
+	tknData := getTokenData(token, indexOfFirstHyphen)
+	tokenTickerLen := len(tknData.ticker)
 
-	tokenTickerLen := len(tokenTicker)
-
-	areTickerAndRandomSequenceInvalid := !esdt.IsTokenTickerLenCorrect(tokenTickerLen) || len(randomSequencePlusNonce) == 0
+	areTickerAndRandomSequenceInvalid := !esdt.IsTokenTickerLenCorrect(tokenTickerLen) || len(tknData.randomSequencePlusNonce) == 0
 	if areTickerAndRandomSequenceInvalid {
 		return tokenKey, 0
 	}
 
-	if len(randomSequencePlusNonce) < esdtTickerNumRandChars+1 {
+	if len(tknData.randomSequencePlusNonce) < esdtTickerNumRandChars+1 {
 		return tokenKey, 0
 	}
 
 	// ALC-1q2w3eX - X is the nonce
-	nonceStr := randomSequencePlusNonce[esdtTickerNumRandChars:]
+	nonceStr := tknData.randomSequencePlusNonce[esdtTickerNumRandChars:]
```

Comment:
is the comment above correct? no need for another hyphen before X? ALC-1q2w3e-X?

Reply:
"Unfortunately", it is correct. Nonce is appended directly to ticker instead of separated by another hyphen.